src/commands.rs

```diff
@@ -2042,93 +2044,118 @@ fn cmd_log(ui: &mut Ui, command: &CommandHelper, args: &LogArgs) -> Result<(), C
+                        template.format(&commit, formatter.as_mut())?;
+                    }
+                }
+                if !buffer.ends_with(b"\n") {
+                    buffer.push(b'\n');
+                }
+                if let Some(diff_format) = diff_format {
+                    let mut formatter = ui.new_formatter(&mut buffer);
+                    show_patch(
+                        formatter.as_mut(),
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+                let node_symbol = if is_checkout { b"@" } else { b"o" };
+                graph.add_node(
+                    &index_entry.position(),
+                    &graphlog_edges,
+                    node_symbol,
+                    &buffer,
                 )?;
             }
-            let node_symbol = if is_checkout { b"@" } else { b"o" };
-            graph.add_node(
-                &index_entry.position(),
-                &graphlog_edges,
-                node_symbol,
-                &buffer,
-            )?;
-        }
-    } else {
-        let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
-            Box::new(revset.iter().reversed())
         } else {
-            Box::new(revset.iter())
-        };
-        for index_entry in iter {
-            let commit = store.get_commit(&index_entry.commit_id())?;
-            template.format(&commit, formatter)?;
-            if let Some(diff_format) = diff_format {
-                show_patch(
-                    formatter,
-                    &workspace_command,
-                    &commit,
-                    matcher.as_ref(),
-                    diff_format,
-                )?;
+            let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
+                Box::new(revset.iter().reversed())
+            } else {
+                Box::new(revset.iter())
+            };
+            for index_entry in iter {
+                let commit = store.get_commit(&index_entry.commit_id())?;
+                template.format(&commit, formatter)?;
+                if let Some(diff_format) = diff_format {
+                    show_patch(
+                        formatter,
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+            }
+        }
+    }
+
+    // Check to see if the user might have specified a path when they intended
+    // to specify a revset.
+    if let [only_path] = args.paths.as_slice() {
+        if only_path == "." {
+            // For users of e.g. Mercurial, where `.` indicates the current commit.
+            ui.write_warn(&format!(
+                "warning: The argument {only_path:?} is being interpreted as a path, but this is \
```

Comment:
Do you think it's better to use `only_path.display()` instead of the debug formatting? I don't know what the difference is in practice.

Reply:
I used the debug formatting here because it'll produce a quoted string, probably suitable to use on the command-line. If we have a shell-quoting library or function already, we could use that.

Reply:
I've used `.display()` in a few places. Maybe we should use debug formatting instead, but that's not important enough that I'm going to bother changing it.